apps/core/secureaction.h

```diff
@@ -0,0 +1,84 @@
+/****************************************************************************
+**
+** Copyright (c) 2018 - 2021 Jolla Ltd.
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SECUREACTION_H
+#define SECUREACTION_H
+
+#include <QObject>
+#include <QDBusAbstractAdaptor>
+
+#include <QJSValue>
+
+#include <memory>
+
+QT_BEGIN_NAMESPACE
+class QDBusPendingCallWatcher;
+QT_END_NAMESPACE
+
+class SecureAction;
+
+class SecureActionAuthenticatorAdaptor : public QDBusAbstractAdaptor
+{
+    Q_OBJECT
+    Q_CLASSINFO("D-Bus Interface", "org.nemomobile.devicelock.client.Authenticator")
+public:
+    explicit SecureActionAuthenticatorAdaptor(SecureAction *action);
+
+public slots:
+    Q_NOREPLY void PermissionGranted(uint method);
+    Q_NOREPLY void Aborted();
+
+private:
+    SecureAction * const m_secureAction;
+};
+
+class SecureAction : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(bool available READ isAvailable NOTIFY availableChanged)
+    Q_PROPERTY(QString message READ message WRITE setMessage NOTIFY messageChanged)
```

Comment:
to be API compat should a requireAuthentication property be added, also?

Reply:
Not needed, it's internal API and require authentication was a just a authentication hasn't been asked for yet flag. It starts true and is cleared when authentication is granted.